pkg/workload/schemachange/operation_generator.go

```diff
@@ -2385,7 +2383,8 @@ func (og *operationGenerator) setColumnDefault(ctx context.Context, tx pgx.Tx) (
 	}
 	// Generated columns cannot have default values.
 	if columnForDefault.generated {
-		stmt.expectedExecErrors.add(pgcode.Syntax)
+		stmt.potentialExecErrors.add(pgcode.Syntax)
```

Comment:
should we only add this to potential errors if `generated` is true? 

I'm mostly just trying to make sure I understand. why should we not always expect the UndefinedObject code here? this is in the if statement where we know we're using a type that doesn't exist, so aren't we always getting that error code? is it a mixed version issue where older versions are still returning SyntaxError even for types that don't exist?

also, what does postgres return if you set a default on a computed column on a type that doesn't exist?

maybe the thing that would help me the most is if the commit message could explain the postgres behavior, the CRDB behavior before https://github.com/cockroachdb/cockroach/pull/127545, and the CRDB behavior after https://github.com/cockroachdb/cockroach/pull/127545.

Reply:
yes -- does the new commit message help? let me know if can do anything else